jobs/cluster-6/templates/bin/post-deploy

```diff
@@ -1,19 +1,46 @@
 #!/bin/bash
+
+function log() {
+  echo "[$(date)] $*"
+}
+
 <% if !spec.bootstrap %>log "this node is not the bootstrap node; nothing to do."
 exit 0
 <% else %>set -eu
 exec >&2
 
+<% if p('redis.tls.enabled') -%>
+# Install and setup stunnel.
+apt install -y redis-tools stunnel4
+
 function redis() {
   local ip=$1 ; shift
+
+  # We need to setup stunnel for TLS connections.
+  cat > /etc/stunnel/stunnel.conf << EOF
+[redis-cli]
+  client = yes
+  accept = 127.0.0.1:8000
+  connect = $ip:<%= p('redis.tls.dual-mode') ? 16379 : 6379 %>
+EOF
+
+  # Restart stunnel to apply changes.
+  kill `cat /var/run/stunnel4.pid` || true
+  stunnel /etc/stunnel/stunnel.conf &> /dev/null
+
+  # Execute the actual redis command.
   (echo "AUTH <%= p('auth.password') %>"; echo $*) \
-    | /var/vcap/packages/redis-6/bin/redis-cli -h $ip \
+    | /var/vcap/packages/redis-6/bin/redis-cli -h 127.0.0.1 -p 8000 \
```

Comment:
So that I'm clear in my understanding - Redis listens to 8000 then stunnel listens on `$ip` and redis port?

Reply:
`stunnel` opens a TLS connection to `$ip` which is the redis server instance, `redis-cli` connects to `stunnel` which is listening to port `8000` to forward commands to the TLS endpoint of the redis server.